### DISCUSSION TITLE
Field Text Array

### DISCUSSION THREAD

**Discussion by HeriYantodotDev** [keystonejs/keystone#8636] [Questions] [answered]
Hi everyone,

I'm currently exploring the KeystoneJS documentation at https://keystonejs.com/docs/fields/overview#scalar-types, and I've noticed that there is no scalar type specifically designated for a text array.

Rather, it appears that input for an array of text would require a custom type, as explained in the documentation's section on backend customization: https://keystonejs.com/docs/guides/custom-fields#backend.

While creating a custom type is one way to handle this, I wanted to inquire if there might be any workarounds or if someone in the community has already developed a custom type for a text array.

I greatly appreciate your assistance and suggestions. Thank you in advance!

**Comment by marekryb** [comment, marked as answer]
Hi,

The simplest solution would be to go with [JSON](https://keystonejs.com/docs/fields/json) field. If you are fine with validating the data in `validateInput` hook, then custom type might not be necessary. But since you already started writing it, I am guessing this is not the case.

The only repo with additional custom types that I know of is this one: https://github.com/keystonejs-contrib/k6-contrib

You did not write what kind of array data you want to store, but it might have an impact on the way forward. Storing text for multiple languages would be good example. There are several ways of doing this (see eg. [this blog post](https://vertabelo.com/blog/multi-language-database-design/)) and none of them is a perfect solution.

**Reply by HeriYantodotDev** [reply]
Thanks for the suggestions @coderua. They're really helpful.
Yup, `json` field will do.
